clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -113,13 +182,23 @@ public void commitSync(final Duration timeout) {
         try {
             commitFuture.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
         } catch (final TimeoutException e) {
-            throw new org.apache.kafka.common.errors.TimeoutException("timeout");
+            throw new org.apache.kafka.common.errors.TimeoutException(
+                     "timeout");
         } catch (final Exception e) {
             // handle exception here
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Get the current subscription.  or an empty set if no such call has
+     * been made.
+     * @return The set of topics currently subscribed to
+     */
+    public Set<String> subscription() {
+        return Collections.unmodifiableSet(new HashSet<>(this.subscriptions.subscription()));
```

Comment:
Why create a new, intermediate `HashSet` instead of just:

```suggestion
        return Collections.unmodifiableSet(this.subscriptions.subscription());
```

Reply:
Does the locking around the subscription happen elsewhere?

Reply:
I think subscription is thread safe~